sdk/appconfiguration/azure-appconfigurationprovider/azure/appconfigurationprovider/_azureappconfigurationprovider.py

```diff
@@ -0,0 +1,217 @@
+            return
+        self._client = AzureAppConfigurationClient(
+            endpoint, credential, user_agent=useragent, headers=headers)
+
+    @staticmethod
+    def __resolve_keyvault_reference(config, key_vault_options, secret_clients):
+        if key_vault_options is None:
+            raise AttributeError(
+                "Key Vault options must be set to resolve Key Vault references.")
+
+        if config.secret_id is None:
+            raise AttributeError(
+                "Key Vault reference must have a uri value.")
+
+        key_vault_identifier = KeyVaultSecretIdentifier(config.secret_id)
+
+        referenced_client = next(
+            (client for client in secret_clients if client.vault_url == key_vault_identifier.vault_url), None)
+
+        if referenced_client is None and key_vault_options.credential is not None:
+            referenced_client = SecretClient(
+                vault_url=key_vault_identifier.vault_url, credential=key_vault_options.credential)
+            secret_clients[key_vault_identifier.vault_url] = referenced_client
+
+        if referenced_client:
+            try:
+                return referenced_client.get_secret(key_vault_identifier.name, version=key_vault_identifier.version)
+            except ResourceNotFoundError as e:
+                raise KeyVaultReferenceError("Key Vault %s does not contain secret %s" % (
+                    key_vault_identifier.vault_url, key_vault_identifier.name))
+            except HttpResponseError as e:
+                raise e
+
+        if key_vault_options.secret_resolver is not None:
+            return key_vault_options.secret_resolver(config.secret_id)
+        raise KeyVaultReferenceError(
+            "No Secret Client found for Key Vault reference %s" % (key_vault_identifier.vault_url))
+
+    @staticmethod
+    def __is_json_content_type(content_type):
+        if not content_type:
+            return False
+
+        content_type = content_type.strip().lower()
+        mime_type = content_type.split(';')[0].strip()
+
+        type_parts = mime_type.split('/')
+        if len(type_parts) != 2:
+            return False
+
+        (main_type, sub_type) = type_parts
+        if main_type != "application":
+            return False
+
+        sub_types = sub_type.split('+')
+        if "json" in sub_types:
+            return True
+
+        return False
+
+    def __trim(self, key):
+        for trim in self._trim_prefixes:
+            if key.startswith(trim):
+                return key[len(trim):]
+        return key
+
+    def __getitem__(self, key):
+        return self._dict[key]
+
+    def __repr__(self):
+        return repr(self._dict)
+
+    def __len__(self):
+        return len(self._dict)
+
+    def copy(self):
+        """
+        Returns a copy of the configuration settings
+
+        type: () -> dict
+        """
+        return self._dict.copy()
+
+    def has_key(self, k):
```

Comment:
`has_key` should not be used in python 3 (it has been removed). If we want to be dict-like, then `in` (as is used in the implementation) should be preferred.

Reply:
Updated to be `in` instead of `has_key`